src/components/UpcomingEventsList.tsx

```diff
@@ -53,31 +60,44 @@ const UpcomingEventsList = () => {
     )
 
     // Add formatted string to display
-    const formattedEvents = orderedEvents.map((event) => {
-      const getDate = (date) =>
-        getLocaleTimestamp(locale! as Lang, dateParse(date).toString(), {})
-
-      const dateRange =
-        event.startDate === event.endDate
-          ? getDate(event.startDate)
-          : `${getDate(event.startDate)} - ${getDate(event.endDate)}`
-
-      const details = `${event.description}`
-
-      return {
-        ...event,
-        date: dateRange,
-        formattedDetails: details,
+    const formattedEvents: Array<FormattedCommunityEvent> = orderedEvents.map(
+      (event) => {
+        const getDate = (date) =>
+          getLocaleTimestamp(locale! as Lang, dateParse(date).toString(), {})
+
+        const dateRange =
+          event.startDate === event.endDate
+            ? getDate(event.startDate)
+            : `${getDate(event.startDate)} - ${getDate(event.endDate)}`
+
+        const details = `${event.description}`
+
+        return {
+          ...event,
+          date: dateRange,
+          formattedDetails: details,
+        }
       }
-    })
-    const groupedEvents = groupBy(formattedEvents, ({ startDate }) => {
-      const start = new Date(startDate.replace(/-/g, "/"))
-      const formatYearMonth = new Intl.DateTimeFormat(locale, {
-        month: "short",
-        year: "numeric",
-      }).format(start)
-      return `${formatYearMonth}`
-    })
+    )
+
+    const groupedEvents = formattedEvents.reduce<MonthGroupedEvents>(
+      (acc, event) => {
+        const start = new Date(event.startDate.replace(/-/g, "/"))
+        const formatYearMonth = new Intl.DateTimeFormat(locale, {
+          month: "short",
+          year: "numeric",
+        }).format(start)
+        const key = `${formatYearMonth}`
+
+        if (!acc[key]) {
+          acc[key] = []
+        }
+
+        acc[key].push(event)
+        return acc
```

Comment:
Agree with comment..

```tsx
`${}`
```

^ Redundant pattern, but we then don't need to assign to `const key` either

```suggestion

        if (!acc[formatYearMonth]) {
          acc[formatYearMonth] = []
        }

        acc[formatYearMonth].push(event)
        return acc
```

or, just name `formatYearMonth` `key` instead